internal/server/iigointernal/executive.go

```diff
@@ -0,0 +1,148 @@
+// Called by orchestration at the end of the turn
+func (e *executive) getTaxMap(islandsResources map[shared.ClientID]shared.Resources) map[shared.ClientID]shared.Resources {
+	e.budget -= serviceCharge
+	result, _ := e.clientPresident.SetTaxationAmount(islandsResources)
+	return result
+}
+
+// broadcastTaxation broadcasts the tax amount decided by the president to all island still in the game.
+func (e *executive) broadcastTaxation(islandsResources map[shared.ClientID]shared.Resources) {
+	e.budget -= serviceCharge
+	taxAmountMap := e.getTaxMap(islandsResources)
+	for _, v := range getIslandAlive() {
+		d := baseclient.Communication{T: baseclient.CommunicationInt, IntegerData: int(taxAmountMap[shared.ClientID(int(v))])}
+		data := make(map[int]baseclient.Communication)
+		data[TaxAmount] = d
+		communicateWithIslands(shared.TeamIDs[int(v)], shared.TeamIDs[e.ID], data)
+	}
+}
+
+// Send Tax map all the remaining islands
+// Called by orchestration at the end of the turn
+func (e *executive) getAllocationRequests(commonPool shared.Resources) map[shared.ClientID]shared.Resources {
+	e.budget -= serviceCharge
+	result, _ := e.clientPresident.EvaluateAllocationRequests(e.ResourceRequests, commonPool)
+	return result
+}
+
+func (e *executive) requestAllocationRequest() {
+	allocRequests := make(map[shared.ClientID]shared.Resources)
+	for _, v := range getIslandAlive() {
+		allocRequests[shared.ClientID(int(v))] = iigoClients[shared.ClientID(int(v))].CommonPoolResourceRequest()
+	}
+	AllocationAmountMapExport = allocRequests
+	e.setAllocationRequest(allocRequests)
+
+}
+
+// replyAllocationRequest broadcasts the allocation of resources decided by the president
+// to all islands alive
+func (e *executive) replyAllocationRequest(commonPool shared.Resources) {
+	e.budget -= serviceCharge
+	allocationMap := e.getAllocationRequests(commonPool)
+	for _, v := range getIslandAlive() {
+		d := baseclient.Communication{T: baseclient.CommunicationInt, IntegerData: int(allocationMap[shared.ClientID(int(v))])}
+		data := make(map[int]baseclient.Communication)
+		data[AllocationAmount] = d
+		communicateWithIslands(shared.TeamIDs[int(v)], shared.TeamIDs[e.ID], data)
+	}
+}
+
+// appointNextSpeaker returns the island id of the island appointed to be speaker in the next turn.
+func (e *executive) appointNextSpeaker(clientIDs []shared.ClientID) shared.ClientID {
+	e.budget -= serviceCharge
+	var election voting.Election
+	election.ProposeElection(baseclient.Speaker, voting.Plurality)
+	election.OpenBallot(clientIDs)
+	election.Vote(iigoClients)
+	return election.CloseBallot()
+}
+
+// withdrawSpeakerSalary withdraws the salary for speaker from the common pool.
+func (e *executive) withdrawSpeakerSalary(gameState *gamestate.GameState) error {
+	var speakerSalary = shared.Resources(rules.VariableMap["speakerSalary"].Values[0])
+	var withdrawError = WithdrawFromCommonPool(speakerSalary, gameState)
+	if withdrawError != nil {
+		e.speakerSalary = speakerSalary
+	}
```

Comment:
erm... I feel like you got the logic about Errors wrong. If there is an error -> assign values speakerSalary ??

If this is intended. Then this is not an Error. Please don't use Errors this way.

Also also I had a look at the WithdrawFromCommonPool function - the error returned is error.Errorf("Not enough in common pool..") but you still give a salary to the Speaker.

You also did this in withdrawJudgeSalary().

Reply:
Thanks for this @E-Mac27 ! The condition should be == . We had made the fix in judiciary but neglected to do it in executive and legislature. Sorted it now :)